core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -121,10 +125,77 @@ protected void doStop() throws Exception {
       }
     };
 
-    NetworkTrafficServerConnector connector = new NetworkTrafficServerConnector(server);
-    connector.addNetworkTrafficListener(new MetricsListener(metrics, "jetty", metricTags));
-    connector.setPort(getConfiguration().getInt(RestConfig.PORT_CONFIG));
-    server.setConnectors(new Connector[]{connector});
+    // http
+    if (config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP) ||
+        config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP_PLUS_HTTPS)) {
+      NetworkTrafficServerConnector connector = new NetworkTrafficServerConnector(server);
+      connector.addNetworkTrafficListener(new MetricsListener(metrics, "jetty", metricTags));
+      connector.setPort(getConfiguration().getInt(RestConfig.PORT_CONFIG));
+      server.addConnector(connector);
+      log.info("Added http connector.");
+    }
+
+    // https
+    if (config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTPS) ||
+        config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP_PLUS_HTTPS)) {
+      SslContextFactory sslContextFactory = new SslContextFactory();
+      // IMPORTANT: the key's CN, stored in the keystore, must match the FQDN. This is a Jetty requirement.
```

Comment:
Doesn't Jetty support `AltSubjectNames`? I would be surprised if it doesn't. We have been telling people about the CN and FQDN for Kafka too, but it turns out that this is actually no longer the recommended way. See https://issues.apache.org/jira/browse/KAFKA-3665?focusedCommentId=15274001&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15274001 for more details.